instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py

```diff
@@ -592,6 +592,49 @@ def test_instrument_connection(self):
         connection2 = dbapi.instrument_connection(self.tracer, connection, "-")
         self.assertIs(connection2.__wrapped__, connection)
 
+    @mock.patch("opentelemetry.instrumentation.dbapi.DatabaseApiIntegration")
+    def test_instrument_connection_kwargs_defaults(self, mock_dbapiint):
+        connection = mock.Mock()
+        # Avoid get_attributes failing because can't concatenate mock
+        connection.database = "-"
+        dbapi.instrument_connection(self.tracer, connection, "-")
+        kwargs = mock_dbapiint.call_args[1]
+        self.assertEqual(kwargs["connection_attributes"], None)
+        self.assertEqual(kwargs["version"], "")
+        self.assertEqual(kwargs["tracer_provider"], None)
+        self.assertEqual(kwargs["capture_parameters"], False)
+        self.assertEqual(kwargs["enable_commenter"], False)
+        self.assertEqual(kwargs["commenter_options"], None)
+        self.assertEqual(kwargs["connect_module"], None)
+
+    @mock.patch("opentelemetry.instrumentation.dbapi.DatabaseApiIntegration")
+    def test_instrument_connection_kwargs_provided(self, mock_dbapiint):
+        mock_tracer_provider = mock.MagicMock()
+        mock_connect_module = mock.MagicMock()
+        connection = mock.Mock()
+        # Avoid get_attributes failing because can't concatenate mock
+        connection.database = "-"
+        dbapi.instrument_connection(
+            self.tracer,
+            connection,
+            "-",
+            connection_attributes={"foo": "bar"},
+            version="test",
+            tracer_provider=mock_tracer_provider,
+            capture_parameters=True,
+            enable_commenter=True,
+            commenter_options={"foo": "bar"},
+            connect_module=mock_connect_module,
+        )
+        kwargs = mock_dbapiint.call_args[1]
+        self.assertEqual(kwargs["connection_attributes"], {"foo": "bar"})
+        self.assertEqual(kwargs["version"], "test")
+        self.assertEqual(kwargs["tracer_provider"], mock_tracer_provider)
```

Comment:
not sure how mocks behave with `==` check, you may want to do `assertIs()` to assert they are the same instance (and on L636)

Reply:
Yes! Changed in [ce86c1e](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3027/commits/ce86c1e3a1080ed574e136fe5cd525af072cc12b)